Automattic/newspack-plugin
# feat(blocks): add meta to core/image blocks

### All Submissions:






### Changes proposed in this Pull Request:

- Added hooks to allow `show_in_rest` post-meta to be added as attributes to `core/image` blocks.
- Added styling for visual feedback during media requests:
   - Individual `image/core` blocks have a gradient animation and `<Spinner />` overlay.
   - "Update" post button is locked, pending media request resolution
- Added fallback logic to credit html to use block attributes if empty and `meta` attribute exists.
- Registered attachment credit post-meta.

These change are necessary for user who use plugins like [Distributor](https://github.com/10up/distributor) and are unable to display credit meta on pulling sites. This PR aims to solve this by adding it the core/image blocks attributes

Asana: 1205909204837811-as-1206457195610270/f

### How to test the changes in this Pull Request:

1. Ensure you have [Distributor](https://github.com/10up/distributor) installed and active
2. You have two sites connected using Distributors External Connections
5. Ensure you have an image (Media Library) where credit post-meta has been added

![Screenshot 2024-02-16 at 16 42 10](https://github.com/Automattic/newspack-plugin/assets/3676975/74f1b469-d954-4a37-ac05-354a8ad7c73e)

6. Create a post, or edit existing one, and add above image to the block editor (`post_content`, then click Save/Update. Once saved, click "Distribute" post.

![Screenshot 2024-02-16 at 16 57 42](https://github.com/Automattic/newspack-plugin/assets/3676975/f1e23a3a-e25a-4631-84bd-03b8a8e5f0ea)

8. Confirm credit is on both the source & destination sites for in-content attachments.

![Screenshot 2024-02-16 at 17 00 50](https://github.com/Automattic/newspack-plugin/assets/3676975/2093b299-e08e-45cf-8e16-d0229742565c)

### Other information:
